.ci/Jenkinsfile

```diff
@@ -115,28 +122,32 @@ pipeline {
   }
   post {
     cleanup {
-      notifyBuildResult()
+      notifyBuildResult(downstreamJobs: itsDownstreamJobs)
     }
   }
 }
 
 def runJob(agentName, buildOpts = ''){
-  def branch = env.BRANCH_NAME
-
-  if (env.CHANGE_ID) {
-    branch = env.CHANGE_TARGET
+  def branch = env.CHANGE_ID?.trim() ? env.CHANGE_TARGET : env.BRANCH_NAME
+  def job
+  try {
+    def quiet = randomNumber(min: 2, max: 15) + 5
```

Comment:
Is there any case where we might want to parametrize the span length? Perhaps in the future we may wish to set a certain repo to have a more or less aggressive retry policy?

Reply:
This is a workaround to add some latency when running a bunch of downstream in parallel at the same time. It's not ideal though and IMO, it should be handled by the infra itself ecosystem, to avoid this weird and unnecessary code.

There were different options, in fact, https://github.com/elastic/apm-integration-testing/blob/e38ce906b1cdae87364bad03975fc31819013c17/.ci/Jenkinsfile#L22 is the programmatic way to avoid one single repo to cope with all the resources from the CI.

Since the retry/sleep workaround, when running the checkout, is now embedded with the shared library, I'd say I can remove this sleep randomness feature as it was an attempt to help with the timeout environmental issue, it did work just a bit by the way :(

Reply:
Makes sense. 👍 Thanks!